### PR TITLE
SNOW-164802 resolve OCSP flakiness and add more logging

### PR DESCRIPTION
**Pull request by sfc-gh-mkeller**
From #413 
This fixes the most often failing flaky tests

It mostly just adds more logging and fixes the tests for the OCSP cache the right way.

### CODE REVIEW

File: test/test_ocsp.py [hunk 1]
```diff
@@ -212,8 +220,7 @@ def test_ocsp_with_bogus_cache_files(tmpdir):
 
     # forces to use the bogus cache file but it should raise errors
     SnowflakeOCSP.clear_cache()
-    ocsp = SFOCSP(
-        ocsp_response_cache_uri='file://' + cache_file_name)
```
Comment (sfc-gh-stakeda) [inline test/test_ocsp.py:216]: nit; maybe we won't need this parameter any longer, then in the future change, can we remove it?

**Review by sfc-gh-stakeda — APPROVED** [review]

File: test/test_ocsp.py (hunk 1 above)
Comment (sfc-gh-mkeller) [reply test/test_ocsp.py:216]: I will give myself a ticket to remove it (SNOW-200287).

Comment (sfc-gh-hchaturvedi) [reply test/test_ocsp.py:216]: Are we absolutely sure that the customer will not need it?

**Review by sfc-gh-hchaturvedi — APPROVED** [review]
Changes look good to me. I suggest we give it a stronger thought that the customer will never need to set an alternate location to the OCSP response cache file before removing that option. If we decide to keep it then we should be looking at fixing the connector to use the user specified location (if set) rather than reverting to a hardcoded location every time.

**Pull request merged**